assignment3/word_search.py

```diff
@@ -0,0 +1,93 @@
+#! /usr/bin/python3.5
+
+import dictionary
+ALL_WORDS = set()
+WORD_DICT = None
+CHAR_2D_ARRAY = None
+
+def findAllWords(row, col, char2dArray, wordDict):
+    """Find all valid words that can be formed in the char2dArray. 
+    Note that CHAR_2D_ARRAY, WORD_DICT and ALL_WORDS are global variables. 
```

Comment:
QAQ A WordSearcher class sounds like a more reasonable choice... I made the global variables when I realized that I was passing around too many variables, and making a class did not occur to me QAQ.